collections/deep_merge.ts

```diff
@@ -71,7 +73,7 @@ export function deepMerge<
 
     const a = record[key] as ResultMember;
 
-    if (!(key in other)) {
+    if (!(hasOwnProperty(other, key))) {
```

Comment:
We're removing `_util/has_own_property.ts` in #1177. Can you use Object.hasOwn instead?

Reply:
Done!